tests/std/tests/P0896R4_ranges_alg_set_difference/test.cpp

```diff
@@ -0,0 +1,127 @@
+                          ranges::less{}, identity{}, identity{})),
+    ranges::set_difference_result<int*, int*>>);
+
+struct instantiator {
+    static constexpr P elements1[] = {{0, 10}, {0, 11}, {0, 12}, {1, 10}, {1, 11}, {3, 10}};
+    static constexpr P elements2[] = {{13, 0}, {14, 0}, {10, 2}, {11, 3}, {12, 3}};
+    static constexpr P expected[]  = {{0, 12}, {1, 10}, {1, 11}};
+
+    template <ranges::input_range R1, ranges::input_range R2, weakly_incrementable O>
+    static constexpr void call() {
+        using ranges::set_difference, ranges::set_difference_result, ranges::equal, ranges::iterator_t, ranges::less;
+
+        constexpr auto osize = ranges::size(elements1) + ranges::size(elements2);
+
+        { // Validate range overload
+            P output[osize]{};
+            R1 range1{elements1};
+            R2 range2{elements2};
+            const same_as<set_difference_result<iterator_t<R1>, O>> auto result =
+                set_difference(range1, range2, O{output}, ranges::less{}, get_first, get_second);
+            assert(result.in == range1.end());
+            assert(result.out.peek() == output + ranges::size(expected));
+            assert(equal(span{output}.first<ranges::size(expected)>(), expected));
+        }
+        { // Validate iterator overload
+            P output[osize]{};
+            R1 range1{elements1};
+            R2 range2{elements2};
+            const same_as<set_difference_result<iterator_t<R1>, O>> auto result = set_difference(range1.begin(),
+                range1.end(), range2.begin(), range2.end(), O{output}, ranges::less{}, get_first, get_second);
+            assert(result.in == range1.end());
+            assert(result.out.peek() == output + ranges::size(expected));
+            assert(equal(span{output}.first<ranges::size(expected)>(), expected));
+        }
+
+        { // Validate range overload, empty range1
+            P output[osize]{};
+            R1 range1{};
+            R2 range2{elements2};
+            const same_as<set_difference_result<iterator_t<R1>, O>> auto result =
+                set_difference(range1, range2, O{output}, ranges::less{}, get_first, get_second);
+            assert(result.in == range1.end());
+            assert(result.out.peek() == output);
+        }
+        { // Validate iterator overload, empty range2
+            P output[osize]{};
+            R1 range1{elements1};
+            R2 range2{};
+            const same_as<set_difference_result<iterator_t<R1>, O>> auto result = set_difference(range1.begin(),
+                range1.end(), range2.begin(), range2.end(), O{output}, ranges::less{}, get_first, get_second);
+            assert(result.in == range1.end());
+            assert(result.out.peek() == output + ranges::size(elements1));
+            assert(equal(span{output}.first<ranges::size(elements1)>(), elements1));
+        }
+    }
+};
+
+template <class Continuation>

```

Comment:
That is three times you use those instantiators. Even if we cannot find more uses for them it would be beneficial to move them to range_algorithm_helpers

Reply:
These are algorithm-specific enough that I prefer that they live in the algorithm-specific test sources.